terraform/aws/components/gp2gp/s3.tf

```diff
@@ -0,0 +1,59 @@
+
+# S3 bucket for storing extracts as cache
+resource "aws_s3_bucket" "gp2gp_extract_cache_bucket" {
+  bucket = "${var.environment_id}-gp2gp-extract-cache-bucket"
```

Comment:
The prefix for resource name is combined in `local.resource_prefix` variable. So the name of the bucket will be `"${local.resource_prefix}-extract_cache_bucket"

Reply:
s3 bucket name cannot contain underscore ( _ ) character, so using hyphen ( - ) in bucket name. local.resource_prefix is also structured with hyphen.

resource_prefix = "${var.project}-${var.environment}-${var.component}"

https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html